sdk/hash/Cargo.toml

```diff
@@ -0,0 +1,39 @@
+[package]
+name = "solana-hash"
+description = "Solana wrapper for the 32-byte output of a hashing algorithm."
+documentation = "https://docs.rs/solana-hash"
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+borsh = { workspace = true, optional = true }
+bs58 = { workspace = true }
+bytemuck = { workspace = true, optional = true }
+bytemuck_derive = { workspace = true, optional = true }
+js-sys = { workspace = true }
+serde = { workspace = true, optional = true }
+serde_derive = { workspace = true, optional = true }
+solana-atomic-u64 = { workspace = true }
+solana-frozen-abi = { workspace = true, optional = true }
+solana-frozen-abi-macro = { workspace = true, optional = true }
+solana-sanitize = { workspace = true }
+thiserror = { workspace = true }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+wasm-bindgen = { workspace = true }
+
+[build-dependencies]
+rustc_version = { workspace = true }
+
+[features]
+borsh = ["dep:borsh"]
+bytemuck = ["dep:bytemuck", "dep:bytemuck_derive"]
+frozen-abi = ["dep:solana-frozen-abi", "dep:solana-frozen-abi-macro"]
+serde = ["dep:serde", "dep:serde_derive"]
```

Comment:
We don't use a `bytemuck` feature for any other crates in the monorepo, and I'm not sure it's really necessary here. At a minimum, can we make it a default feature?

Reply:
Ok so the compile times for small crates like this are dominated by the syn, quote and proc-macro2 crates, which bytemuck uses.

However thiserror also uses those crates, so I've now added a commit to replace thiserror with a cleaned-up version of what rust-analyzer expands the `derive(Error)` to. As you can see it's very simple.

I've also opened a [PR](https://github.com/anza-xyz/agave/pull/2165) to similarly remove thiserror from solana-sanitize, which solana-hash depends on. Once that's merged, solana-hash will not pull in syn and friends by default.

I would rather not make bytemuck a default feature because the impact on build times is quite substantial